janusgraph-core/src/main/java/org/janusgraph/diskstorage/keycolumnvalue/inmemory/BufferPageUtils.java

```diff
@@ -0,0 +1,78 @@
+// Copyright 2017 JanusGraph Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.janusgraph.diskstorage.keycolumnvalue.inmemory;
+
+import java.io.DataInputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This is a collection of static methods to help read a BufferPage from a stream it was previously dumped to
+ */
+public class BufferPageUtils {
+    public static BufferPage readPage(DataInputStream in) throws IOException {
+        int numEntries = in.readInt();
+        //System.out.println("page numentries is " + numEntries);
+
+        if (numEntries > 0) {
+            int[] index = new int[numEntries];
+            //NOTE: again, no way to read int[] as a block of bytes apparently... relying on buffering
+            for (int i = 0; i < index.length; i++) {
+                index[i] = in.readInt();
+            }
+
+            int dataLength = in.readInt();
+            //System.out.println("page data size is " + dataLength);
+
+            byte[] data = new byte[dataLength];
+            readWholeArray(in, data);
+
+            return new BufferPage(index, data);
+        } else
+            return new BufferPage(BufferPage.EMPTY_INDEX, BufferPage.EMPTY_DATA);
+    }
+
+    public static void readWholeArray(DataInputStream in, byte[] data) throws IOException {
+        int offset = 0;
+        do {
+            int bytesRead = in.read(data, offset, data.length - offset);
+            if (bytesRead < 0) {
+                throw new IllegalStateException("Premature end of file while reading, expected to read " + data.length);
+            }
+            offset += bytesRead;
+        }
+        while (offset < data.length);
+    }
+
+    public static SharedEntryBuffer readFrom(DataInputStream in) throws IOException {
+        int numPages = in.readInt();
+
+        //System.out.println("number of pages in column store is " + numPages);
```

Comment:
Could you remove this or use a logger with the level trace?

Reply:
used logger